File: src/go/finished/sum_of_digits_of_string_after_convert.go
package leetcode

import (
	"strconv"
	"strings"
)

// https://leetcode.com/problems/sum-of-digits-of-string-after-convert

func getLucky(s string, k int) int {
	sb := strings.Builder{}
	for _, ch := range s {
		sb.WriteString(strconv.Itoa(int(ch) - 96))
	}
	digits := sb.String()

	for i := 0; i < k; i++ {
		sum := 0
		for _, d := range digits {
			digit, _ := strconv.Atoi(string(d))
			sum += digit
		}
		digits = strconv.Itoa(sum)
	}
	res, _ := strconv.Atoi(digits)
	return res
}

File: src/python/finished/sum_of_digits_of_string_after_convert.py
# https://leetcode.com/problems/sum-of-digits-of-string-after-convert

class Solution:
    def getLucky(self, s: str, k: int) -> int:
        new_s = []
        for c in s:
            new_s.append(str(ord(c)-ord('a')+1))
        new_s = "".join(new_s)
        
        res = new_s
        for _ in range(k):
            tmp = 0
            for c in res:
                tmp += int(c)
            res = str(tmp)
        return int(res)